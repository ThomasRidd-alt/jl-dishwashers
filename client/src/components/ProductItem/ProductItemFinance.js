const ProductItemFinance = props => {

    const getGuaranteeInfo = () => {
        return <div className={'item-page__finance-guarantee jl-body-text'}>{props.includedServices.map(service => <p
            key={service}>{service}</p>)}</div>
    }

    const getDisplaySpecialOffer = () => {
        if (props.displaySpecialOffer) {
            return <p className={'item-page__finance-special-offer jl-highlight-text'}>Special
                Offer: {props.displaySpecialOffer}</p>
        } else {
            return null
        }
    }

    return (
        <section className={'dishwasher-section-container item-page__sidebar-column-section'}>
            <p className={'item-page__finance-price jl-body-text-large'}>Â£{props.price}</p>
            {getDisplaySpecialOffer()}
            {getGuaranteeInfo()}
        </section>
    )
};

export default ProductItemFinance