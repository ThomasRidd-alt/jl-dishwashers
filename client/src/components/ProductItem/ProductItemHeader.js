import chevron from '../../images/chevron-left.svg'
import {Link} from "react-router-dom";

const ProductItemHeader = props => {
    return (
        <header className={'item-page__header'}>
            <div className={'dishwasher-width-container'}>
                <div className={'item-page__header-items-container'}>
                    <Link to={'/'}><img className={'item-page__header-back chevron'} src={chevron} alt={'back'}/></Link>
                    <h1 className={'item-page__header-title jl-heading'}>{props.title}</h1>
                </div>
            </div>
        </header>
    )
};

export default ProductItemHeader