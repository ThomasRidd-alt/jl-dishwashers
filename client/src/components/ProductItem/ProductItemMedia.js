import Carousel from "../Carousel/Carousel";

const ProductItemMedia = props => {
    return (
        <section className={'dishwasher-section-container item-page__main-column-section'}>
            <div className={'item-page__images-container'}>
                <Carousel images={props.images.urls}/>
            </div>
        </section>
    )
};

export default ProductItemMedia