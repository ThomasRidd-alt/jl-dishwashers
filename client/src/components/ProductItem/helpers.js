const getInfoSummary = info => {
    const open = info.indexOf("<p>")
    const close = info.indexOf("</p>")
    if (open >= 0 && close > open) {
        return info.substring(open + 3, close)
    }
    return info
}

export {getInfoSummary}