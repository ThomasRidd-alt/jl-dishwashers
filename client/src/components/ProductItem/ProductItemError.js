import ProductItemHeader from "./ProductItemHeader";
import React from "react";

const ProductItemError = () => {

    return (<>
        <ProductItemHeader title={'There was an error'}/>
        <main>
            <p>There was an error finding this product</p>
        </main>
    </>)
}

export default ProductItemError;