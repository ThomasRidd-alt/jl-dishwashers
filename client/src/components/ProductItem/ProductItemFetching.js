import ProductItemHeader from "./ProductItemHeader";
import React from "react";

const ProductItemFetching = () => {

    return (<>
        <ProductItemHeader title={'Loading'}/>
        <main>
            <p>Fetching data</p>
        </main>
    </>)
}

export default ProductItemFetching;