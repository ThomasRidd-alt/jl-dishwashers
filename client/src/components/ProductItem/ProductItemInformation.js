import React, {Component} from "react";
import chevronRight from '../../images/chevron-right.svg'
import chevronUp from '../../images/chevron-up.svg'
import {getInfoSummary} from "./helpers";

class ProductItemInformation extends Component {

    constructor(props) {
        super(props)
        this.code = props.code;
        this.productInformation = props.productInformation;
        this.infoSummary = getInfoSummary(this.productInformation);
        this.state = {isOpen: false};
    }

    renderAccordion() {
        return (
            <div className={'item-page__accordion'}>
                <div className={`closed-panel ${this.state.isOpen ? 'visually-hidden' : ''}`} aria-hidden={'true'}>
                    <div className={'item-page__product-summary jl-body-text '}
                         dangerouslySetInnerHTML={{__html: this.infoSummary}}/>
                    <div className={`item-page__accordion-button open-button`}
                         onClick={() => this.setState({isOpen: !this.state.isOpen})}>
                        <p className={'item-page__accordion-caption'}>Read more</p>
                        <img className={'chevron'} src={chevronRight} alt={''}/>
                    </div>
                </div>
                <div className={`open-panel ${this.state.isOpen ? '' : 'visually-hidden'}`}>
                    <div className={'item-page__product-information jl-body-text '}
                         dangerouslySetInnerHTML={{__html: this.productInformation}}/>
                    <div className={`item-page__accordion-button close-button`}
                         onClick={() => this.setState({isOpen: !this.state.isOpen})}
                         aria-hidden={'true'}>
                        <p className={'item-page__accordion-caption'}>Read less</p>
                        <img className={'chevron'} src={chevronUp} alt={''}/>
                    </div>
                </div>
            </div>
        )
    }

    render() {
        return (
            <section className={'dishwasher-section-container item-page__main-column-section'}>
                <h2 className={'item-page__section-heading jl-heading'}>Product information</h2>
                <p className={'jl-body-text'}>Product code: {this.code}</p>
                {this.renderAccordion()}
            </section>
        )
    }
}

export default ProductItemInformation