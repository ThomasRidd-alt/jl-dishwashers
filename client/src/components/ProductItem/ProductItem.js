import ProductItemHeader from "./ProductItemHeader";
import ProductItemMedia from "./ProductItemMedia";
import ProductItemFinance from "./ProductItemFinance";
import ProductItemInformation from "./ProductItemInformation";
import ProductItemSpecification from "./ProductItemSpecification";
import React from "react";

const ProductItem = props => {
    let {product} = props;
    return (<>
        <ProductItemHeader title={product.title}/>
        <main>
            <div className={'dishwasher-width-container item-page__container'}>
                <ProductItemMedia images={product.media.images}/>
                <ProductItemFinance price={product.price.now}
                                    displaySpecialOffer={product.displaySpecialOffer}
                                    includedServices={product.additionalServices.includedServices}/>
                <ProductItemInformation code={product.code} productInformation={product.details.productInformation}/>
                <ProductItemSpecification specification={product.details.features[0].attributes}/>
            </div>
        </main>
    </>)
}

export default ProductItem;