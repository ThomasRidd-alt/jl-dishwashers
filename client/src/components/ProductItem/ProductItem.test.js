import React from "react";
import {shallow} from 'enzyme';
import {mockProductItemData} from "../../data/productData";
import ProductItem from "./ProductItem";

describe('ProductItem', () => {
    const props = {
        product: mockProductItemData
    }

    let component = shallow(<ProductItem {...props}/>);

    it('renders properly', () => {
        expect(component).toMatchSnapshot()
    })

})