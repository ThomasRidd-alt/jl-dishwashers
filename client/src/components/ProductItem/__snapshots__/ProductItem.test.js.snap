// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem renders properly 1`] = `ShallowWrapper {}`;
